AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefault.m

```diff
@@ -221,6 +230,7 @@ - (void)setEnabled:(BOOL)isEnabled andDeleteDataOnDisabled:(BOOL)deleteData {
 
 #if !TARGET_OS_OSX
 - (void)applicationWillTerminate:(__unused UIApplication *)application {
+  OSAtomicOr32Barrier(1, & _hasEnteredApplicationWillTerminate);
 
   // Block logs queue so that it isn't killed before app termination.
```

Comment:
Can't we just set the boolean in the log queue synchronously with dispatch_sync? Trying to simplify because we are introducing quite some complexity for just this scenario. Need to find a tradeoff or postpone the fix.

Reply:
It won't make it thread safe

Reply:
Right because by the time it's running some more logs may be queued asynchronously. Though, we can still set the bool before an empty dispatch sync. In this place we are guaranteed to run in the main thread so a dispatch sync should be safe. Of course unless the code dispatch_sync to the main thread but that would be bad.